apis/apiextensions/v1alpha1/infrastructuredefinition_types.go

```diff
@@ -33,6 +34,17 @@ type InfrastructureDefinitionSpec struct {
 	// user of the defined kind.
 	ConnectionSecretKeys []string `json:"connectionSecretKeys,omitempty"`
 
+	// DefaultCompositionRef refers to the Composition resource that will be used
+	// in case no composition selector is given.
+	// +optional
+	DefaultCompositionRef *corev1.ObjectReference `json:"defaultCompositionRef,omitempty"`
+
+	// EnforcedCompositionRef refers to the Composition resource that will be used
+	// by all composite instances whose schema is defined by this definition.
+	// +optional
+	// +immutable
+	EnforcedCompositionRef *corev1.ObjectReference `json:"enforcedCompositionRef,omitempty"`
```

Comment:
Can we avoid proliferating the use of ObjectReference here? See the rationale in https://github.com/crossplane/crossplane-runtime/issues/49.

Reply:
I've been using it just because most meta methods that work with object references support it, including ours. Though I agree with the rationale behind it.

Reply:
I actually wanted to refactor the tooling in crossplane-runtime that uses `ObjectReference`, like set/get composition reference functions. However, the types we have don't exactly meet this use case.

`v1alpha1.TypedReference` could make sense but `apiVersion` and `kind` fields are required in that one, which doesn't for us right now since we have only `Composition` kind and it's verbose to enforce people to write its `apiVersion` and `kind` right now.

`v1alpha1.Reference` has only `Name`, which works for now but I know that we'll want different `kind`s of `Composition`s in the future and that will require churn on Go code level (not on API level really, since `name` field will stay).

I will go with `v1alpha1.Reference` for now but save that refactoring for later.